project-icp/bee-pollinator-app
# Use model changes to update results view

Previously, making a modification to a scenario while initially polling for results would result in an error. This PR fixes this bug by switching from using a promise to using model events to update the UI. See commit message for more details.

#### Testing
* Check that the bug cannot be replicated on this branch.
* Cause a timeout error to occur and check that timeout error message is shown. You can do this by changing the constant at  https://github.com/project-icp/bee-pollinator-app/blob/develop/src/icp/js/src/core/models.js#L95 
* Cause a network error to occur by turning off the network in developer tools and check that the generic error message is shown.
* With results showing, trigger a model run and check that the current results are shown while polling, and that the new results appear. 
* Repeat above while logged in.
* Repeat above in the compare view, although there is a pre-existing bug that may come up documented in https://github.com/project-icp/bee-pollinator-app/issues/115

Connects #105
